# **Chinook Database Analysis Assignment**

## **Scenario:**

You are a data analyst at **MusicWave Inc.**, a company that specializes in music distribution and sales. Your company uses the Chinook database to manage its music collection, sales, and customer information. Recently, your manager has tasked you with analyzing the database to extract valuable insights that can help the company make informed business decisions.

## **Introduction to the Chinook Database:**

The Chinook database is a sample database that represents a digital media store. It includes tables for:
- **Artists**: Information about the music artists.
- **Albums**: Information about the albums, each linked to an artist.
- **Tracks**: Information about individual tracks, including which album they belong to.
- **Customers**: Information about the customers.
- **Invoices**: Information about sales transactions.
- **InvoiceLines**: Detailed information about each item in an invoice.
- **Genres**: Information about the genres of the tracks.
- **MediaTypes**: Information about the types of media.
- **Employees**: Information about the employees.

The database schema allows for complex queries to analyze various aspects of the music store's operations.

## **Getting Started:**

### **Download the Chinook Database**

1. Download the Chinook SQLite database from [here](https://github.com/lerocha/chinook-database/raw/master/ChinookDatabase/DataSources/Chinook_Sqlite.sqlite).

2. Upload the `Chinook_Sqlite.sqlite` file to your Google Colab environment.

### **Setting Up the Environment**

Run the following cells to set up your environment and connect to the Chinook database.

```python
# Install sqlite3 library (if not already installed)
!pip install sqlite3

# Import necessary libraries
import sqlite3
import pandas as pd

# Connect to the SQLite database
conn = sqlite3.connect('Chinook_Sqlite.sqlite')


In [6]:
import sqlite3
import pandas as pd

# Connect to the SQLite database
conn = sqlite3.connect('Chinook_Sqlite.sqlite')

# Execute a query
query = "SELECT * FROM sqlite_master where type='table'"

df = pd.read_sql_query(query, conn)
df[['type','name']]



type           name
0   table          Album
1   table         Artist
2   table       Customer
3   table       Employee
4   table          Genre
5   table        Invoice
6   table    InvoiceLine
7   table      MediaType
8   table       Playlist
9   table  PlaylistTrack
10  table          Track

In [9]:
# Read the first five rows from the Album table
df = pd.read_sql_query("SELECT * FROM album limit 5", conn)
print(df)

   AlbumId                                  Title  ArtistId
0        1  For Those About To Rock We Salute You         1
1        2                      Balls to the Wall         2
2        3                      Restless and Wild         2
3        4                      Let There Be Rock         1
4        5                               Big Ones         3


### **Question 1: List All Tables**

**Scenario:**  
Your manager, Alex, is new to the team and is trying to understand the structure of the Chinook database. Alex asks you to provide a list of all the tables in the database to get an overview of the available data.

**Task:**  
Write a query to list all tables in the Chinook database.

In [11]:
query1 = "SELECT name FROM sqlite_master WHERE type='table';"
df1 = pd.read_sql_query(query1, conn)
print(df1)

             name
0           Album
1          Artist
2        Customer
3        Employee
4           Genre
5         Invoice
6     InvoiceLine
7       MediaType
8        Playlist
9   PlaylistTrack
10          Track


### **Question 2: Retrieve Top 10 Records from the Albums Table**

**Scenario:**  
Alex wants to familiarize himself with the kind of albums stored in the database. He asks you to fetch a sample of the album records to get a better idea of the data.

**Task:**  
Write a query to retrieve the top 10 records from the Albums table.

---


In [12]:
query2 = "SELECT * FROM Album LIMIT 10;"
df2 = pd.read_sql_query(query2, conn)
print(df2)

   AlbumId                                  Title  ArtistId
0        1  For Those About To Rock We Salute You         1
1        2                      Balls to the Wall         2
2        3                      Restless and Wild         2
3        4                      Let There Be Rock         1
4        5                               Big Ones         3
5        6                     Jagged Little Pill         4
6        7                               Facelift         5
7        8                         Warner 25 Anos         6
8        9         Plays Metallica By Four Cellos         7
9       10                             Audioslave         8



### **Question 3: Count the Number of Tracks per Genre**

**Scenario:**  
To plan a marketing campaign, Alex needs to know which genres are most represented in the music store. He asks you to count the number of tracks available for each genre.

**Task:**  
Write a query to count the number of tracks per genre.

---




In [13]:
query3 = """
SELECT Genre.Name, COUNT(Track.TrackId) AS TrackCount
FROM Genre
JOIN Track ON Genre.GenreId = Track.GenreId
GROUP BY Genre.Name
ORDER BY TrackCount DESC;
"""
df3 = pd.read_sql_query(query3, conn)
print(df3)

                  Name  TrackCount
0                 Rock        1297
1                Latin         579
2                Metal         374
3   Alternative & Punk         332
4                 Jazz         130
5             TV Shows          93
6                Blues          81
7            Classical          74
8                Drama          64
9             R&B/Soul          61
10              Reggae          58
11                 Pop          48
12          Soundtrack          43
13         Alternative          40
14         Hip Hop/Rap          35
15   Electronica/Dance          30
16               World          28
17         Heavy Metal          28
18    Sci Fi & Fantasy          26
19      Easy Listening          24
20              Comedy          17
21          Bossa Nova          15
22     Science Fiction          13
23       Rock And Roll          12
24               Opera           1


### **Question 4: Calculate the Total Sales by Country**

**Scenario:**  
Alex is interested in understanding the revenue distribution across different countries. He asks you to calculate the total sales generated from each country.

**Task:**  
Write a query to calculate the total sales by country.

In [14]:
query4 = """
SELECT BillingCountry, SUM(Total) AS TotalSales
FROM Invoice
GROUP BY BillingCountry
ORDER BY TotalSales DESC;
"""
df4 = pd.read_sql_query(query4, conn)
print(df4)

    BillingCountry  TotalSales
0              USA      523.06
1           Canada      303.96
2           France      195.10
3           Brazil      190.10
4          Germany      156.48
5   United Kingdom      112.86
6   Czech Republic       90.24
7         Portugal       77.24
8            India       75.26
9            Chile       46.62
10         Ireland       45.62
11         Hungary       45.62
12         Austria       42.62
13         Finland       41.62
14     Netherlands       40.62
15          Norway       39.62
16          Sweden       38.62
17          Poland       37.62
18           Italy       37.62
19         Denmark       37.62
20       Australia       37.62
21       Argentina       37.62
22           Spain       37.62
23         Belgium       37.62



### **Question 5: Identify the Top 5 Customers by Total Purchase Amount**

**Scenario:**  
To improve customer relationship management, Alex wants to identify the top spenders. He asks you to find out who the top 5 customers are based on their total purchase amount.

**Task:**  
Write a query to identify the top 5 customers by total purchase amount.

---


In [15]:
query5 = """
SELECT Customer.FirstName, Customer.LastName, SUM(Invoice.Total) AS TotalSpent
FROM Customer
JOIN Invoice ON Customer.CustomerId = Invoice.CustomerId
GROUP BY Customer.CustomerId
ORDER BY TotalSpent DESC
LIMIT 5;
"""
df5 = pd.read_sql_query(query5, conn)
print(df5)

  FirstName    LastName  TotalSpent
0    Helena        Holý       49.62
1   Richard  Cunningham       47.62
2      Luis       Rojas       46.62
3  Ladislav      Kovács       45.62
4      Hugh    O'Reilly       45.62



### **Question 6: Find the Most Popular Artist by Total Track Sales**

**Scenario:**  
Alex wants to identify the most popular artist in the store based on track sales. He asks you to find out which artist has the highest total track sales.

**Task:**  
Write a query to find the most popular artist by total track sales.

---


In [16]:
query6 = """
SELECT Artist.Name, COUNT(InvoiceLine.TrackId) AS TracksSold
FROM Artist
JOIN Album ON Artist.ArtistId = Album.ArtistId
JOIN Track ON Album.AlbumId = Track.AlbumId
JOIN InvoiceLine ON Track.TrackId = InvoiceLine.TrackId
GROUP BY Artist.Name
ORDER BY TracksSold DESC
LIMIT 1;
"""
df6 = pd.read_sql_query(query6, conn)
print(df6)

          Name  TracksSold
0  Iron Maiden         140



### **Question 7: Determine the Average Invoice Total by Customer Country**

**Scenario:**  
To understand the purchasing behavior of customers from different countries, Alex asks you to determine the average invoice total for each country.

**Task:**  
Write a query to determine the average invoice total by customer country.

---



In [17]:
query7 = """
SELECT BillingCountry, AVG(Total) AS AverageInvoice
FROM Invoice
GROUP BY BillingCountry
ORDER BY AverageInvoice DESC;
"""
df7 = pd.read_sql_query(query7, conn)
print(df7)

    BillingCountry  AverageInvoice
0            Chile        6.660000
1          Ireland        6.517143
2          Hungary        6.517143
3   Czech Republic        6.445714
4          Austria        6.088571
5          Finland        5.945714
6      Netherlands        5.802857
7            India        5.789231
8              USA        5.747912
9           Norway        5.660000
10         Germany        5.588571
11          France        5.574286
12          Sweden        5.517143
13        Portugal        5.517143
14          Brazil        5.431429
15          Canada        5.427857
16          Poland        5.374286
17           Italy        5.374286
18         Denmark        5.374286
19       Australia        5.374286
20       Argentina        5.374286
21  United Kingdom        5.374286
22           Spain        5.374286
23         Belgium        5.374286


### **Question 8: Analyze Employee Sales Performance**

**Scenario:**  
Alex wants to evaluate the performance of the sales team. He asks you to analyze the sales performance of each employee.

**Task:**  
Write a query to analyze the sales performance of each employee.

---



In [18]:
query8 = """
SELECT Employee.FirstName, Employee.LastName, SUM(Invoice.Total) AS TotalSalesHandled
FROM Employee
JOIN Customer ON Employee.EmployeeId = Customer.SupportRepId
JOIN Invoice ON Customer.CustomerId = Invoice.CustomerId
GROUP BY Employee.EmployeeId;
"""
df8 = pd.read_sql_query(query8, conn)
print(df8)

  FirstName LastName  TotalSalesHandled
0      Jane  Peacock             833.04
1  Margaret     Park             775.40
2     Steve  Johnson             720.16


### **Question 9: Identify Trends in Monthly Sales**

**Scenario:**  
Alex wants to identify sales trends over time. He asks you to analyze the monthly sales data to spot any trends.

**Task:**  
Write a query to identify trends in monthly sales.

---



In [19]:
query9 = """
SELECT strftime('%Y-%m', InvoiceDate) AS Month, SUM(Total) AS MonthlySales
FROM Invoice
GROUP BY Month
ORDER BY Month;
"""
df9 = pd.read_sql_query(query9, conn)
print(df9)

      Month  MonthlySales
0   2021-01         35.64
1   2021-02         37.62
2   2021-03         37.62
3   2021-04         37.62
4   2021-05         37.62
5   2021-06         37.62
6   2021-07         37.62
7   2021-08         37.62
8   2021-09         37.62
9   2021-10         37.62
10  2021-11         37.62
11  2021-12         37.62
12  2022-01         52.62
13  2022-02         46.62
14  2022-03         44.62
15  2022-04         37.62
16  2022-05         37.62
17  2022-06         37.62
18  2022-07         37.62
19  2022-08         37.62
20  2022-09         36.63
21  2022-10         37.62
22  2022-11         37.62
23  2022-12         37.62
24  2023-01         37.62
25  2023-02         37.62
26  2023-03         37.62
27  2023-04         51.62
28  2023-05         42.62
29  2023-06         50.62
30  2023-07         37.62
31  2023-08         37.62
32  2023-09         37.62
33  2023-10         37.62
34  2023-11         23.76
35  2023-12         37.62
36  2024-01         37.62
37  2024-02 

### **Question 10: Recommend Marketing Strategies Based on Data Insights**

**Scenario:**  
Based on the insights you have gathered from the previous queries, Alex asks you to recommend at least two marketing strategies that MusicWave Inc. should consider to increase sales and customer satisfaction.

**Task:**  
Based on your analysis, write a brief recommendation for two marketing strategies that could help MusicWave Inc. increase sales and improve customer satisfaction.



---


**To:** Alex, Management Team

**From:** Data Analysis Department

**Subject: Data-Driven Marketing Strategies for MusicWave Inc.**

Based on the analysis of the Chinook database, I recommend the following two strategies to drive revenue growth and enhance customer loyalty:

1. The "Rock Legend" Artist Spotlight Campaign
Data Insight: Our analysis shows that Rock is the dominant genre with 1,297 tracks (Question 3), and Iron Maiden is our top-performing artist with 140 tracks sold (Question 6).

Recommendation: MusicWave Inc. should launch a targeted "Rock Legend" campaign. Since the USA is our largest market by total sales ($523.06, Question 4), we should deploy targeted email marketing and homepage features specifically for our Rock catalog in North America. By bundling top-selling Iron Maiden tracks with lesser-known Rock artists, we can increase the average transaction value while leveraging our most popular genre.

2. High-Value Customer Retention Program (Tiered Loyalty)
Data Insight: We have identified a core group of high-spending customers, led by Helena Holý and Richard Cunningham, who spend significantly above the average (Question 5). Additionally, countries like Chile and the Czech Republic show the highest "Average Invoice" totals (Question 7), suggesting these customers are willing to spend more per visit.

Recommendation: Implement a "MusicWave VIP" Loyalty Program specifically targeting our top 5% of spenders. High-value customers in the Czech Republic and Chile should receive exclusive "First Access" to new releases or personalized discounts based on their purchase history. This strategy shifts the focus from simple acquisition to high-value retention, ensuring our most profitable customers remain satisfied and continue their high-spending behavior.